plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt

```diff
@@ -167,29 +169,40 @@ private fun mapLicense(
  */
 internal suspend fun mapSnippetFindings(
     rawResults: RawResults,
+    snippetsLimit: Int,
     issues: MutableList<Issue>,
     detectedLicenseMapping: Map<String, String>,
     snippetChoices: List<SnippetChoice>,
     snippetLicenseFindings: MutableSet<LicenseFinding>
 ): Set<SnippetFinding> {
     val remainingSnippetChoices = snippetChoices.toMutableList()
-    val allFindings = mutableSetOf<SnippetFinding>()
-
-    rawResults.listSnippets.map { (file, rawSnippets) ->
-        rawSnippets.mapSnippetFindingsForFile(
-            file,
-            rawResults.snippetMatchedLines,
-            issues,
-            detectedLicenseMapping,
-            snippetChoices,
-            remainingSnippetChoices,
-            snippetLicenseFindings
-        )
-    }.collect {
-        allFindings += it
+    val results = mutableSetOf<SnippetFinding>()
+    var runningSnippetCount = 0
+
+    coroutineScope {
```

Comment:
Is this construct with `coroutineScope` and `launch` required? I would assume that the flow can be collected directly in this `suspend` function. Or is this done because of the `cancel()` to enforce the limit?

Reply:
Yes it is required: `cancel()` is a function of the `coroutineScope`, not from the `Flow`.

Reply:
Sorry, I am confused. Could you elaborate what is the exact purpose of this construct? And would the `takeWhile` approach I suggested in the other comment be an alternative?

Reply:
`cancel()` was needed to stop the coroutine collecting the emitted values.
I removed it for now as `transformWhile` achieves this requirement.